frontend/views/containers/user-settings/AppLogs.vue

```diff
@@ -108,8 +128,28 @@ export default ({
         }
       })
     },
-    downloadLogs () {
-      sbp('appLogs/download', this.$refs.linkDownload)
+    async downloadOrShareLogs () {
+      const actionType = this.ephemeral.useWebShare ? 'share' : 'download'
+      const isDownload = actionType === 'download'
+
+      try {
+        await sbp('appLogs/downloadOrShare',
+          actionType,
+          isDownload ? this.$refs.linkDownload : undefined
+        )
+      } catch (err) {
+        const errorDisplay = isDownload
+          ? L('Failed to download the app logs. {reportError}', LError(err))
+          : L('Failed to share the app logs. {reportError}', LError(err))
+
+        console.error(`AppLogs.vue downloadOrShareLogs() '${actionType}' action error:`, err)
+        this.$refs.errBanner.danger(errorDisplay)
+      }
+    },
+    checkWebShareAvailable () {
+      this.ephemeral.useWebShare = Boolean(navigator.share) &&
+        window.matchMedia('(hover: none) and (pointer: coarse)').matches &&
+        window.matchMedia('screen and (max-width: 1199px)').matches
```

Comment:
Whether to show `Share` instead of `Download` on button is checked here. So even when the app is viewed on a mobile browser with no desktop screen size (which is `>= 1200px` in our app), if `navigator.share` is not supported. The button will still say `Download`.